File: sources/nim/_06_char_count.nim
# task 06 char_count — expected output: 10000000
# build: nim c -d:release -o:prog _06_char_count.nim    run: ./prog

import std/strutils

let text = repeat("abcdefghij", 10_000_000)
var count: int64 = 0
for ch in text:
  if ch == 'a': discard
  elif ch == 'e': discard
  elif ch == 'h': inc count
  else: discard
echo count

File: sources/julia/06_char_count.jl
# task 06 char_count — expected output: 10000000
# build: julia 06_char_count.jl    run: julia 06_char_count.jl

function main()
    text = repeat("abcdefghij", 10000000)
    h = UInt8('h')
    count = Int64(0)
    n = ncodeunits(text)
    for i in 1:n
        if codeunit(text, i) == h
            count += 1
        end
    end
    println(count)
end

main()
